File: backend/app/schemas/document.py
"""Document schemas."""
from typing import Optional

from pydantic import BaseModel


class CreateDocumentSchema(BaseModel):
    """Schema for creating a document."""

    client_id: str
    name: str
    document_type: str
    file_url: Optional[str] = None
    file_size: Optional[int] = None
    mime_type: Optional[str] = None


class DocumentResponseSchema(BaseModel):
    """Schema for document response."""

    id: str
    client_id: str
    name: str
    document_type: str
    status: str
    file_url: Optional[str] = None
    analysis_result: Optional[dict] = None
    created_at: str
    updated_at: str

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  clientId: string;
  name: string;
  type: DocumentType;
  status: DocumentStatus;
  fileUrl?: string;
  fileSize?: number;
  mimeType?: string;
  analysisResult?: DocumentAnalysis;
  createdAt: string;
  updatedAt: string;
}

export type DocumentType =
  | 'tax_return'
  | 'bank_statement'
  | 'investment_statement'
  | 'insurance_policy'
  | 'other';

export type DocumentStatus = 'pending' | 'processing' | 'analyzed' | 'error';

export interface DocumentAnalysis {
  summary: string;
  keyPoints: string[];
  extractedData: Record<string, unknown>;
  confidence: number;
}
